src/common/src/types/mod.rs

```diff
@@ -596,6 +597,33 @@ impl !PartialOrd for ScalarRefImpl<'_> {}
 pub type Datum = Option<ScalarImpl>;
 pub type DatumRef<'a> = Option<ScalarRefImpl<'a>>;
 
+/// An adapter type to support upstream data types that don't have `ScalarImpl` implementations
+#[derive(Debug)]
+pub enum DatumAdapter {
+    Datum(Datum),
+    Uuid(uuid::Uuid),
+}
+
+impl ToSql for DatumAdapter {
```

Comment:
This is not `common` but specific to interaction with upstream (external) systems. Furthermore, this `ToSql` is only for PostgreSQL upstream.